pkg/controllers/user/helm/controller.go

```diff
@@ -124,10 +117,28 @@ func (l *Lifecycle) Remove(obj *v3.App) (*v3.App, error) {
 			return obj, err
 		}
 	}
+	tempDir, err := ioutil.TempDir("", "helm-")
+	if err != nil {
+		return obj, err
+	}
+	defer os.RemoveAll(tempDir)
+	kubeConfigPath, err := l.writeKubeConfig(obj, tempDir)
+	if err != nil {
+		return obj, err
+	}
+	// try three times and succeed
+	start := time.Second * 1
+	for i := 0; i < 3; i++ {
+		if err := helmDelete(kubeConfigPath, obj); err != nil {
```

Comment:
You never break?  I know 3 is better than 1, but maybe if it succeeds we break :)